### PR TITLE
Add `_recorded_sales` meta to COT

### LINKED ISSUE #32843: [COT] Add `_recorded_sales` column in custom table
`_recorded_sales` is also one of the meta keys that we store for pretty much every order, however, we missed including it in the initial table structure.

This issue scopes adding this meta as a column in order operational data table. We should:
- [ ] Add a new column in the operational data table with the appropriate data type.
- [ ] Add its support for migration in `WPPostToOrderOpTableMigrator`.
-  [ ] Modify all relevant unit tests to add a check for this property.

### PR DESCRIPTION
**Pull request by jorgeatorres**
### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds `recorded_sales` to the meta in the operational data table and updates migration and tests code to account for that.

Closes #32843.

### How to test the changes in this Pull Request:

1. Attempt to migrate a batch or an order and see that despite the metadata `_recorded_sales` (stored as "yes"/"no") being available in `wp_postmeta` for orders, no similar record exists in COT tables.
2. Re-create COT tables by going to WC > Status > Tools and using "Create the custom orders tables" or "Delete the custom orders tables" as necessary.
3. Re-attempt the migration from step 1 and confirm that `recorded_sales` is now a column in the `wc_order_operational_data` table and correctly registers the value from the original post meta.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by jorgeatorres** [comment]
@vedanshujain: One question I had when working on this... when changing the db schema for COT at this stage (i.e. it hasn't been officially released), do we need to care about updating tables that might exist in setups of early testers/adopters or can we assume that the final schema will come later?

File: plugins/woocommerce/tests/php/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationControllerTest.php [hunk 1]
```diff
@@ -311,6 +311,10 @@ private function assert_order_op_data_is_migrated( $order ) {
 		$this->assertEquals( (float) $order->get_shipping_total(), (float) $db_order_op_data->shipping_total_amount );
 		$this->assertEquals( (float) $order->get_discount_tax(), (float) $db_order_op_data->discount_tax_amount );
 		$this->assertEquals( (float) $order->get_discount_total(), (float) $db_order_op_data->discount_total_amount );
+		$this->assertEquals(
+			$order->get_data_store()->get_download_permissions_granted( $order ),
```
Comment (vedanshujain) [inline plugins/woocommerce/tests/php/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationControllerTest.php]: Seems like this data store method needs to be changed?

**Review by vedanshujain — CHANGES_REQUESTED** [review]
Requesting minor change in the unit test, but otherwise looks good

File: plugins/woocommerce/tests/php/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationControllerTest.php (hunk 1 above)
Comment (jorgeatorres) [reply plugins/woocommerce/tests/php/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationControllerTest.php]: 🤦. You're absolutely right.

Comment (jorgeatorres) [reply plugins/woocommerce/tests/php/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationControllerTest.php]: Fixed.

**Comment by jorgeatorres** [comment]
@vedanshujain: Thanks for the review. I've fixed the issue you've mentioned. There are some phpcs errors that are affecting the result of the checks, but those are unrelated to the files in this PR.

**Review by vedanshujain — APPROVED** [review]

**Pull request merged**